kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/coroutines/CallbackCoroutinesSpecificationTest.kt

```diff
@@ -0,0 +1,112 @@
+package com.sksamuel.kotest.engine.coroutines
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.engine.TestEngineLauncher
+import io.kotest.engine.listener.NoopTestEngineListener
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldStartWith
+import kotlinx.coroutines.CoroutineName
+import kotlin.coroutines.coroutineContext
+
+/**
+ * This test is meant to be the definite source of truth for the behavior of coroutines and callbacks in kotest.
+ */
+class CallbackCoroutinesSpecificationTest : FunSpec() {
```

Comment:
Could we launch the test engine once and then have each test just check for its respective variable?

Reply:
Done